### PR TITLE
FOGL-7855 : Error handling implemented to take care of reconfig requests

### PR DESCRIPTION
**Pull request by gnandan**

### CODE REVIEW

File: kafka.cpp [hunk 1]
```diff
@@ -36,6 +38,31 @@ static void dr_msg_cb(rd_kafka_t *rk, const rd_kafka_message_t *rkmessage, void
...
 
+/**
+ *  Callback to handle errors
+ */
+static void error_cb(rd_kafka_t *rk, int err, const char *reason, void *opaque)
+{
+    rd_kafka_resp_err_t kafkaError = (rd_kafka_resp_err_t) err;
+	switch (kafkaError)
+	{
+		case RD_KAFKA_RESP_ERR__TRANSPORT:
```
Comment (AmandeepArora) [inline kafka.cpp:49]: You can probably combine some of the cases here.

File: kafka.cpp [hunk 2]
```diff
@@ -36,6 +38,31 @@ static void dr_msg_cb(rd_kafka_t *rk, const rd_kafka_message_t *rkmessage, void
...
+			Logger::getLogger()->error("Kafka : %s : %s ", rd_kafka_err2str(kafkaError), reason );
+			break;
+		case RD_KAFKA_RESP_ERR__RESOLVE:
+			Logger::getLogger()->warn("Kafka : %s : %s ", rd_kafka_err2str(kafkaError), reason );
+			break;
+		default:
+			Logger::getLogger()->error("Kafka : %s : %s ", rd_kafka_err2str(kafkaError), reason );
+			break;
+	}
+}
```
Comment (AmandeepArora) [inline kafka.cpp:64]: After processing error, m_error should be reset to false again, here or before the next transmission, is that done somewhere?

File: kafka.cpp (hunk 1 above)
Comment (gnandan) [reply kafka.cpp:51]: Fixed

File: kafka.cpp [hunk 3]
```diff
@@ -18,6 +18,8 @@
 using namespace	std;
 using namespace rapidjson;
 
+static bool m_error = false;
```
[the remark this answers is not in the document]
Comment (gnandan) [reply kafka.cpp]: Fixed

File: kafka.cpp (hunk 2 above)
Comment (gnandan) [reply kafka.cpp:66]: Fixed

File: kafka.cpp (hunk 1 above)
Comment (praveen-garg) [reply kafka.cpp:51]: I think Aman meant to combine the "all brokers down" and "transport" errors cases, as they have same statements execution.

File: plugin.cpp [hunk 4]
```diff
@@ -37,13 +37,15 @@ static const char *default_config = QUOTE({
...
-		"default": "localhost:9092,kafka.local:9092"
+		"default": "localhost:9092,kafka.local:9092",
+		"mandatory": "true"
 		},
 	"topic": {
 		"description": "The topic to send reading data on",
 		"order": "2",
 		"displayName": "Kafka Topic",
-		"type": "string", "default": "Fledge"
+		"type": "string", "default": "Fledge",
```
Comment (praveen-garg) [inline plugin.cpp:47]: Move default to next line (lint/format fix)

**Review by praveen-garg — DISMISSED** [review]
Now Shutdown is smooth?! 

If Kafka broker comes up after a while it recovers and start to send? Or that is beyond the scope of this Jira?

**Comment by gnandan** [comment]
> Now Shutdown is smooth?!
> 
> If Kafka broker comes up after a while it recovers and start to send? Or that is beyond the scope of this Jira?

It's taken care in this JIRA

**Review by AmandeepArora — APPROVED** [review]

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**